tasks/main.yml

```diff
@@ -1,7 +1,10 @@
 ---
 - include_tasks: autodetect.yml
-  # We need to know the engine and emulator if we're creating any new VMs.
-  when: libvirt_vms | rejectattr('state', 'eq', 'absent') | list
+  # We don't need to know the engine and emulator if we're not creating any new
+  # VMs.
+  when: >-
+    (libvirt_vms | selectattr('state', 'defined')
+       | selectattr('state', 'eq', 'absent') | list) != libvirt_vms
```

Comment:
I don't follow this change. Could you explain?

Also, the allowed states are present and absent.

Reply:
I see, it's the defined filter rather than equalto defined.